src/Symfony/Component/DependencyInjection/Util/ReflectionHelper.php

```diff
@@ -0,0 +1,67 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\DependencyInjection\Util;
+
+use Go\ParserReflection\ReflectionClass;
+
+/**
+ * ReflectionHelper provides method for native or static reflection of class.
+ *
+ * @author Martin Hasoň <martin.hason@gmail.com>
+ */
+class ReflectionHelper
+{
+    private static $reflections = array();
+
+    /**
+     * Returns reflection instance for class.
+     *
+     * @param $class Full name of the class
+     *
+     * @return \ReflectionClass
+     */
+    public function getReflection($class)
+    {
+        if (isset(self::$reflections[$class])) {
+            return self::$reflections[$class];
+        }
+
+        if (class_exists($class, false) || interface_exists($class, false) || trait_exists($class, false)) {
+            return $this->getNativeReflection($class);
+        }
+
+        if (class_exists(ReflectionClass::class)) {
```

Comment:
This check could be costly when the class is missing, should be done in the constructor IMO